lib/pf/api.pm

```diff
@@ -566,10 +566,9 @@ sub register_node : Public {
     my ($class, %postdata )  = @_;
     my @require = qw(mac pid);
     my @found = grep {exists $postdata{$_}} @require;
-    return unless pf::util::validate_argv(\@require,  \@found);
+    return (0) unless pf::util::validate_argv(\@require,  \@found);
```

Comment:
remove the `(0)` from the 4 places you added it since it creates a different behavior than all the other API methods.

Other than that, it will be good to merge